superset-frontend/plugins/plugin-chart-echarts/src/pinterest-utils/tooltip.ts

```diff
@@ -0,0 +1,330 @@
+    };
+  };
+
+  getDeltaTableRows(
+    params: CallbackDataParams[],
+    xIndex: number,
+  ): DeltaTableTooltipRow[] {
+    const { pinterestDeltaTableColumns } = this.formData;
+    const deltaTableColumns = this.deltaTableColumns.filter(
+      column =>
+        !PERCENT_CHANGE_COLUMNS.includes(column) ||
+        pinterestDeltaTableColumns.includes(column),
+    );
+    const rows = [
+      {
+        seriesId: 'delta-table-header',
+        columns: deltaTableColumns.map(column => ({
+          element: 'th',
+          style: this.getCellStyle(column),
+          data: column,
+          key: column,
+        })),
+      },
+    ] as DeltaTableTooltipRow[];
+    params.forEach(param => {
+      const deltaTableData = this.getDeltaTableData(
+        (param.value as number[])[xIndex],
+        param.seriesId!,
+        deltaTableColumns,
+      );
+      const newRowColumns = deltaTableColumns.map(column => {
+        const columnData = deltaTableData[column];
+        let color;
+        let data = columnData ?? '-';
+        if (column === DeltaTableColumn.Metric) {
+          data = param.marker + escape(columnData?.toString());
+        } else if (
+          PERCENT_CHANGE_COLUMNS.includes(column) &&
+          columnData != null
+        ) {
+          data += '%';
+          if ((columnData as number) > 0) {
+            color = this.theme.colors.success.dark1;
+            data += DIRECTION_SYMBOL[DeltaDirection.Up];
+          } else if ((columnData as number) < 0) {
+            color = this.theme.colors.error.dark1;
+            data += DIRECTION_SYMBOL[DeltaDirection.Down];
+          }
+        }
+        return {
+          element: 'td',
+          style: this.getCellStyle(column, color),
+          data,
+          key: column,
+        };
+      });
+      const newRow = {
+        seriesId: param.seriesId!,
+        columns: newRowColumns,
+      };
+      rows.push(newRow);
+    });
+    return rows;
+  }
+
+  getTooltipFormatter() {
+    const { richTooltip, tooltipSortByMetric, orientation } = this.formData;
+
+    const [xIndex, yIndex] =
+      orientation === OrientationType.Horizontal ? [1, 0] : [0, 1];
+
+    return (initialParams: TooltipPositionCallbackParams) => {
+      const focusedSeries = this.getFocusedSeries();
+      let params: CallbackDataParams[] = richTooltip
+        ? (initialParams as CallbackDataParams[])
+        : [initialParams as CallbackDataParams];
+      if (tooltipSortByMetric) {
+        params = orderBy(params, [
+          ({ value }: CallbackDataParams) => -1 * (value as number[])[yIndex],
+          ['desc'],
+        ]) as CallbackDataParams[];
+      }
+      const deltaTableRows = this.getDeltaTableRows(params, xIndex);
+      const xValue = (params[0].value as number[])[xIndex];
+
+      return `
+        <span style="font-weight: 700">${this.timeFormatter(xValue)}</span>
+        <br />
+        <table>
+          ${deltaTableRows
```

Comment:
we may create some help function like `createTableRow` and `createTableColumn` to make this more readable.

so it can look something like. 
```
deltaTableRows.map(({ seriesId, columns }) =>  
  createTableRow(seriesId, columns, seriesId === focusedSeries)  
).join(''); ```

also I found `contentStyle` appears several times, can we make it reusable?

or we can create a react component for it by passing the deltaTableRows. if it requires a general html node, can use the ReactDOM.render to render it

Reply:
Added the helper functions, thanks! For `contentStyle`, I'm just copying the logic from the other files in this repo because I want to minimize the code changes we're making to the rest of the repo (so that version upgrades don't get too messy)